packages/grafana-ui/src/components/Collapse/CollapsableSection.tsx

```diff
@@ -41,6 +41,9 @@ export const CollapsableSection = ({
   const [open, toggleOpen] = useState<boolean>(isOpen);
```

Comment:
can you rename this to like `internalOpenState` just to make it super clear please 🙏

Reply:
Also might be a good idea to not call `toggleOpen(!open);` if it `isControlled`?

Reply:
I tried to be as non-breaking change as possible. There could be cases that both props are passed and `toggleOpen` still needs to be called.
- you pass `isOpen` for initial rendering to be open
- you pass `onToggle` for tracking something
I know after your research this scenario shouldn't happen but I'm not sure

Reply:
also, does it make sense to do something like

` onToggle?.(isControlled ? !isOpen : !internalOpenState);`

?

Reply:
`toggleOpen` is terribly named and confusing - `toggleOpen` is the setState function from `useState`. When the component is controlled, we probably shouldn't update state inside it.

Reply:
I think a good idea is to also rename the `isOpen` prop into `defaultOpen`.

Reply:
changes applied

Reply:
@Clarity-89 I'll see if there's some typescript-magic we can do to help communicate this